### ISSUE TITLE
Tip Accessibility

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#5971]
Currently keyboard only users can't access tooltips that are placed on non-interactive elements. Tooltips are opened when their associated element receives focus or is moused over. If a tooltip is placed on an element that doesn't receive focus a keyboard user has no way to know that a tooltip is there or access it. It doesn't feel right to make anything with a tooltip tab-able, I think this would cause confusion but it is worth doing some research on this to find out what the correct accessibility behavior is for tooltips for keyboard only users.

Related slack discussion: https://grommet.slack.com/archives/C09QQEG69/p1645541071830789

**Comment by ShimiSun** [comment]
Transcript from the slack conversation:
> I’ve always learned that Tooltips should be discoverable by both mouse and keyboard users - is there something that we need to do in our theme to make it so that drop targets always have a tab-index or do we need to create and re-export our own wrapped version?
You can see in this story that the only drop target that captures tab presses is the button child: https://storybook.grommet.io/?path=/story/controls-tip-children--children

**Comment by ShimiSun** [comment]
Thank you for filing this issue, I think this issue is raising a valid question regarding the fundamental behavior of the Tip component that is important to clarify.
I'll start with sharing some of the resolutions from a primilanry research regarding the accessibility of tooltips:
1.  [The tooltip never receives focus. Focus stays on the owning element](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role)
2.  [Focusable elements should have interactive semantics](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#focusable_elements_should_have_interactive_semantics)

Quick deduction from following these two best practices resolutions: 
We keep the focus on the owning element of the tooltip (not on the tooltip itself) => the owning element of the tooltip should be an interactive element in order to be focusable and tabbable.
On Grommet, we do thrive to follow best practices on our core implementation of components, hence, we have not supported a non-interactive elements with keyboard focus as it contradicts the fundamentals of focusable elements behavior.

Our best advice in this case will be to revisit your designs and consider whether it follows the accessibility best practices when designing [tooltips](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#accessibility_concerns):
> If the information is important enough for a tooltip, isn't it important enough to always be visible?

> Instead of using tooltips and hiding important information, consider writing clear, succinct, always visible descriptions.

If you are still convinced that you should proceed with a non-interactive owning element for the Tip - Grommet is modular enough to allow it (but not to recommend it). In this case the caller to the Tip component can use a plain Button (with no styling and no action) as the owning interactive element, to allow the tip to be visible during tabbing - PLEASE be mindful that this is an ill advice since it does contradict accesibility best practices and might result other accessibility remifications.
We recommend to challenge your designs to not use tooltips on non-interactive elements.

Feel free to let us know if you have any follow up questions or need advice on your designs. Good luck.

**Comment by ShimiSun** [comment]
Thank you for filing this issue, I think this issue is raising a valid question regarding the fundamental behavior of the Tip component that is important to clarify.
I'll start with sharing some of the resolutions from preliminary research regarding the accessibility of tooltips:
1.  [The tooltip never receives focus. Focus stays on the owning element](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role)
2.  [Focusable elements should have interactive semantics](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#focusable_elements_should_have_interactive_semantics)

Quick deduction from following these two best practices resolutions: 
We keep the focus on the owning element of the tooltip (not on the tooltip itself) => the owning element of the tooltip should be an interactive element in order to be focusable and tabbable.
On Grommet, we do thrive to follow best practices on our core implementation of components, hence, we have not supported non-interactive elements with keyboard focus as it contradicts the fundamentals of focusable elements behavior.

Our best advice, in this case, will be to revisit your designs and consider whether it follows the accessibility best practices when designing [tooltips](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#accessibility_concerns):
> If the information is important enough for a tooltip, isn't it important enough to always be visible?

> Instead of using tooltips and hiding important information, consider writing clear, succinct, always visible descriptions.

If you are still convinced that you should proceed with a non-interactive owning element for the Tip - Grommet is modular enough to allow it (but not to recommend it). In this case, the caller to the Tip component can use a plain Button (with no styling and no action) as the owning interactive element (the Tip child), to allow the tip to be visible during tabbing - PLEASE be mindful that this is a piece of ill advice since it does contradict accessibility best practices and might result in other accessibility ramifications.
We recommend challenging your designs to not use tooltips on non-interactive elements.

Feel free to let us know if you have any follow-up questions or need advice on your designs. Good luck.
